lib/internal/url.js

```diff
@@ -388,7 +388,7 @@ Object.defineProperties(URL.prototype, {
       }, options);
       const ctx = this[context];
       var ret = ctx.scheme;
-      if (ctx.host !== null) {
+      if (ctx.hostname !== null) {
```

Comment:
This should also be `ctx.host !== null`. The `hostname` change only applies below.

Reply:
You're right, `URLContext` uses `null`:
```javascript
const { URL } = require('url');

const myURL = new URL('https://nodejs.org/en/');

const kContext = Object.getOwnPropertySymbols(myURL)[0];

console.log(myURL[kContext]);
```
```
URLContext {
  flags: 400,
  scheme: 'https:',
  username: '',
  password: '',
  host: 'nodejs.org',
  port: null,
  path: [ 'en', '' ],
  query: null,
  fragment: null }
```